metabase/metabase
# Add ability to turn a dataset back into a question

This PR makes it possible to turn datasets back into saved questions using a new "Dataset management" section in the details sidebar.

### Moderation

The "Dataset management" section is going to have a "Verify this dataset" button. I'd prefer to implement that in a different PR. This is how it looks now:

<img width="1578" alt="moderation" src="https://user-images.githubusercontent.com/17258145/140544234-2ef2d4c5-e49d-4b5f-afd9-39d38aa7c6b7.png">

### To Verify

1. Follow the steps from #18702 to create a dataset
2. Click the dataset title to open the details sidebar
3. There should be a new "Dataset management" section
4. Click the "Turn back into a saved question" button
5. The dataset should become a saved question now
    * the "Dataset management" section should disappear
    * the dataset (aka waffle) icon should appear at the top of the sidebar amongst other action icons
7. You should see a toast message saying "This is a question now" at the bottom left of the screen
8. Click the "undo" button on the toast, the question should become a dataset again

### Demo

https://user-images.githubusercontent.com/17258145/140544271-1715881a-96f6-4396-93cc-be9cfe3ab901.mov
